tooling/minimize/tests/ui.rs

```diff
@@ -1,20 +1,30 @@
 use std::path::PathBuf;
 
 use ui_test::color_eyre::eyre::Result;
+use ui_test::dependencies::DependencyBuilder;
+use ui_test::spanned::Spanned;
 use ui_test::{
     run_tests_generic, status_emitter, CommandBuilder, Config, Format, Mode, OutputConflictHandling,
 };
 
 fn cfg(path: &str, mode: Mode) -> Config {
     let mut program = CommandBuilder::rustc();
     program.program = PathBuf::from(env!("CARGO_BIN_EXE_minimize"));
-    Config {
-        mode,
+    let mut config = Config {
         program,
-        dependencies_crate_manifest_path: Some(PathBuf::from("./tests/deps/Cargo.toml")),
         out_dir: PathBuf::from(env!("CARGO_TARGET_TMPDIR")).join("ui"),
         ..Config::rustc(path)
-    }
+    };
+
+    config.comment_defaults.base().mode = Spanned::dummy(mode).into();
+    config.comment_defaults.base().set_custom(
+        "dependencies",
+        DependencyBuilder {
+            crate_manifest_path: "./tests/deps/Cargo.toml".into(),
+            ..Default::default()
+        },
+    );
```

Comment:
I am a bit surprised to see the dependency builder set via `comment_defaults`... and then `set_custom`? It says that's for custom flags but we don't have a custom flag, do we? Is `dependencies`  a magic flag name that is recognized by ui_test?

Reply:
the name doesn't matter, you can name it anything you want. The name only matters for flags that include their own comment parsers that may need to modify an existing item.

Reply:
Now I'm even more confused. So we add a custom flag that's not a flag and we give it a name that's not used for parsing and then somehow that adds a dependency builder...?

Reply:
maybe I should name them "settings". these are the same things as the explicitly named fields on `Revisioned`. E.g. `edition` is not a field anymore, but just the `"edition"` map key. Ideally we wouldn't have a key at all, and just use a set instead of a map, but that makes lookup for overwriting harder.

Reply:
Ah so I guess ideally this would be one of these type-indexed `Any` maps where one can insert and looking things using their type as the key?

Reply:
yes, I may give this another attempt in the future, but for now, a stringly keyed map is just simpler

Reply:
Okay I see, thanks for explaining!